app/main.py

```diff
@@ -1,3 +1,79 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        else:
+            raise TypeError("Unsupported operand for:+ operation")
+
+    def __iadd__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            raise TypeError("Unsupported operand for for:+= operation")
+        return self
+
+    def __mul__(self, other: int | float) -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(self.km * other)
+        else:
+            raise TypeError("Unsupported operand for:* operation")
+
+    def __truediv__(self, other: (int, float)) -> "Distance":
+        if isinstance(other, (int, float)):
+            result = self.km / other
+            return Distance(round(result, 2))
+        else:
+            raise TypeError("Unsupported operand for:/ operation")
```

Comment:
do u need to use 'else' here ?

Reply:
@YuriiKindrat 
I've done all my code without else TypeError.
but another student didn't want to accept this task without it

Reply:
@mikhailyemets U could discuss this with them, so they will approve if u are right )